### PR TITLE
Keep current section displayed & VS Mobile UI

### PR DESCRIPTION
**Pull request by MatthiasReumann**
## Changes 
- Update Mobile UI 
- New button `(Follow Sections)` to automatically switch pages.
- Move all video-section code to `video-sections.ts` and add TypeDoc-Style comments

### Mobile UI
image

**Minimized View**
image

### CODE REVIEW

**Review by alexanderstephan — CHANGES_REQUESTED** [review]
I really like the overall design and the code is also much more tidy now. :+1: 
I found some problems, however some of them could be just due to my local setup, but I wanted to bring them up, just to be sure.

- For me the Follow section button missing on mobile. Can you reproduce this?
- Can we keep the margin under every section of the watch page? The spacer is somehow missing now. Also we might want to go back to the Video Sections heading with the camera icon for consistency. Afterwards indentation is also weird for me, see e.g. the alignment of the description. Might be mixed up due to the merge with #523 :thinking: 
Production:
grafik
Now:
grafik
- I like how the update of the follow button, but I find it a bit hard to see. Maybe it can look similar to the `SHOW MORE` for video descriptions and align to the right end of the video sections?
- E.g. for lecture 3 of FPV I keep getting this error. Can you reproduce this?
grafik This seems to happen after doing some edit to the lecture. :thinking: Then, no video sections are displayed anymore.
- Not sure if it should be done in this PR, but when you create a video section and now worker is available to create the preview, the server just crashes for me. Maybe we can handle that a bit more gracefully?

**Comment by MatthiasReumann** [comment]
> I really like the overall design and the code is also much more tidy now. 👍 I found some problems, however some of them could be just due to my local setup, but I wanted to bring them up, just to be sure.
> 
> * For me the Follow section button missing on mobile. Can you reproduce this?
> * Can we keep the margin under every section of the watch page? The spacer is somehow missing now. Also we might want to go back to the Video Sections heading with the camera icon for consistency. Afterwards indentation is also weird for me, see e.g. the alignment of the description. Might be mixed up due to the merge with #523 🤔
>   Production:
>   grafik
>   Now:
>   grafik
> * I like how the update of the follow button, but I find it a bit hard to see. Maybe it can look similar to the `SHOW MORE` for video descriptions and align to the right end of the video sections?
> * E.g. for lecture 3 of FPV I keep getting this error. Can you reproduce this?
>   grafik This seems to happen after doing some edit to the lecture. 🤔 Then, no video sections are displayed anymore.
> * Not sure if it should be done in this PR, but when you create a video section and now worker is available to create the preview, the server just crashes for me. Maybe we can handle that a bit more gracefully?

1. There is no `Follow Sections` on mobile devices. On Mobile devices I renamed it to `Minimize` which basically is a minimised version of follow sections with just one section.
2. Yup, will definitely fix this. Thanks for pointing out! 👍🏻 
3. That's actually a great idea. Will update.
4. Will look into that. Since `elForSpot` isn't in sections-alpine-code, the error might found somewhere else. 
5. I think this should not be that much work. Will fix that immediately.

Thanks ❤️

**Comment by MatthiasReumann** [comment]
@alexanderstephan 

Deployed the changes to the testserver: https://test.live.mm.rbg.tum.de/w/airplay/498 (lecture with some sections)

Unfortunately, I can't reproduce the bug/error (4) in Safari, Chrome, or Firefox. Can you give me a detailed description on how you produced the error?

**Comment by alexanderstephan** [comment]
> @alexanderstephan
> 
> Deployed the changes to the testserver: https://test.live.mm.rbg.tum.de/w/airplay/498 (lecture with some sections)
> 
> Unfortunately, I can't reproduce the bug/error (4) in Safari, Chrome, or Firefox. Can you give me a detailed description on how you produced the error?

So what I do is on a local server:
1. Create new video sections for a lecture that doesn't have one yet.
2. Go to the watch page of that lecture.
3. Then the video sections don't load and the error occurs.
However, for lectures that do have video sections already, this works perfectly fine. I assume this has to do something with local setups.

Maybe this happens because `<template x-for="(s, i) in vs.getList()" :key="s.fileID">` somehow needs a valid file id, but no one is generated in a local setup. I think this fine for now since it works in production.

File: web/template/partial/stream/video-sections.gohtml [hunk 1]
```diff
@@ -1,39 +1,59 @@
 {{define "videosections"}}
-    <div class = "flex px-2">
-        <div x-data="{sections: new watch.VideoSections('{{.ID}}')}"
-             x-init="await sections.fetch()" class="group relative">
-            <h6 class="font-bold text-3 pb-3 pt-2"><i class="fa-solid fa-film mr-1"></i>Sections</h6>
+    <div x-data="{client: new watch.VideoSectionClient('{{.ID}}')}" class="px-2#">
```
Comment (alexanderstephan) [inline web/template/partial/stream/video-sections.gohtml]:
```suggestion
    <div x-data="{client: new watch.VideoSectionClient('{{.ID}}')}">
```

**Review by alexanderstephan — CHANGES_REQUESTED** [review]
Looks great! :100: 

Just one more question: Is it intended that the follow sections button only becomes visible when switching the page? I would have expected it to return once I click on a video section manually again.

**Comment by MatthiasReumann** [comment]
> > @alexanderstephan
> > Deployed the changes to the testserver: https://test.live.mm.rbg.tum.de/w/airplay/498 (lecture with some sections)
> > Unfortunately, I can't reproduce the bug/error (4) in Safari, Chrome, or Firefox. Can you give me a detailed description on how you produced the error?
> 
> So what I do is on a local server:
> 
> 1. Create new video sections for a lecture that doesn't have one yet.
> 2. Go to the watch page of that lecture.
> 3. Then the video sections don't load and the error occurs.
>    However, for lectures that do have video sections already, this works perfectly fine. I assume this has to do something with local setups.
> 
> Maybe this happens because `<template x-for="(s, i) in vs.getList()" :key="s.fileID">` somehow needs a valid file id, but no one is generated in a local setup. I think this fine for now since it works in production.

I think might be right that this is a problem in a local setup. I'd like to look deeper into that, however, in another PR. OK for you?

**Review by alexanderstephan — APPROVED** [review]
Looks great, thanks! :heart:

**Pull request merged**